README.md

```diff
@@ -121,15 +121,75 @@ but it doesn't traverse the object recursively converting nested items:
 ```
 
 
-## Public API
+## API
+
+### Extending `itemadapter`
+
+This package allows to handle arbitrary item classes, by implementing an adapter interface:
+
+_class `itemadapter.adapter.AdapterInterface(item: Any)`_
+
+Abstract Base Class for adapters. An adapter that handles a specific type of item should
+inherit from this class and implement the abstract methods defined on it. `AdapterInterface`
+inherits from [`collections.abc.MutableMapping`](https://docs.python.org/3/library/collections.abc.html#collections.abc.MutableMapping),
+so all methods from the `MutableMapping` class should be implemented as well.
+
+* _class method `is_item(item: Any) -> bool`_
+
+    Return `True` if the adapter can handle the given item, `False` otherwise
+
+* _method `get_field_meta(self, field_name: str) -> MappingProxyType`_
+
+    Return metadata for the given field name, if available
```

Comment:
Maybe we can extend this a bit more. I expect users to wonder what is the purpose of metadata, what kind of data they can have here, and how to create a MappingProxyType object.